### PR TITLE
update modsecurity packages

### PR DESCRIPTION
**Pull request by Izorkin**
###### Motivation for this change
Update libmodsecurity to 3.0.6
Update nginx modsecurity module to 1.0.2
Add OWASP ModSecurity Core Rule Set

cc @ajs124 @dasJ @Mic92

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [x] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/servers/http/nginx/modules.nix [hunk 1]
```diff
@@ -209,10 +209,10 @@ in
       name = "modsecurity-nginx";
       owner = "SpiderLabs";
       repo = "ModSecurity-nginx";
-      rev = "v1.0.1";
-      sha256 = "0cbb3g3g4v6q5zc6an212ia5kjjad62bidnkm8b70i4qv1615pzf";
+      rev = "v1.0.2";
+      sha256 = "sha256-UXiitc3jZlgXlCsDPS+xEFLNRVgRbn8BCCXUEqAWlII=";
     };
-    inputs = [ pkgs.curl pkgs.geoip pkgs.libmodsecurity pkgs.libxml2 pkgs.lmdb pkgs.yajl ];
+    inputs = [ pkgs.curl pkgs.geoip pkgs.libmodsecurity.dev pkgs.libxml2 pkgs.lmdb pkgs.yajl ];
```
Comment (Mic92) [inline pkgs/servers/http/nginx/modules.nix:215]: Why does this one needs dev but the others don't?

Comment (Izorkin) [reply pkgs/servers/http/nginx/modules.nix:215]: Don't need to include `dev`?. Can I remove it? There will be no problems with finding the header files  later?

Comment (Mic92) [reply pkgs/servers/http/nginx/modules.nix:215]: If there were, we would have them with the other packages as well, but normally dev outputs are used by default.

Comment (Mic92) [reply pkgs/servers/http/nginx/modules.nix:215]: I think you can drop it if this is list is just passed to `buildInputs`.

Comment (Izorkin) [reply pkgs/servers/http/nginx/modules.nix]: Fixed.

File: pkgs/top-level/all-packages.nix [hunk 2]
```diff
@@ -7718,6 +7718,8 @@ with pkgs;
 
   modsecurity_standalone = callPackage ../tools/security/modsecurity { };
 
+  modsecurity-csr = callPackage ../tools/security/modsecurity-csr { };
```
Comment (dasJ) [inline pkgs/top-level/all-packages.nix:7721]: Typo in both the attr and the path. It's `crs`, not `csr`

Comment (Izorkin) [reply pkgs/top-level/all-packages.nix]: Oops, fixed. Thanks.

**Comment by Izorkin** [comment]
cc @ajs124

**Review by stigtsp — APPROVED** [review]
LGTM

 - Builds OK on x86_64-linux,
 - Sources are legit and latest 

Result of `nixpkgs-review pr 149591` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>3 packages built:</summary>
  <ul>
    <li>libmodsecurity</li>
    <li>modsecurity-crs</li>
    <li>tengine</li>
  </ul>
</details>

**Pull request merged**